### ISSUE TITLE
Handle dependencies between webppl headers

### ISSUE THREAD

**Issue by hawkrobe** [probmods/webppl#318]
In our basic package system, you can include a list of webppl header files in your `package.json` file:

```javascript
{
  "name": "testModule",
  "webppl": {
    "wppl" : ["header1.wppl", "header2.wppl"]
  }
}
```

For many refactoring schemes, you'll want to define variables or functions in one header and use them in the other, e.g.  
```javascript
// header1.wppl
var a = 1;
```
```javascript
// header2.wppl
var b = a + 1
```
and then `b` is assigned to `2` in your main code.

Currently, whether this works or not depends entirely on the order that you write the two files in `package.json`. The order written above works, because `header1.wppl` is appended first, but if you swap the order, e.g. 
```javascript
{
  "name": "testModule",
  "webppl": {
    "wppl" : ["header2.wppl", "header1.wppl"]
  }
}
```
then `b` is `NaN` in the main code.

The way we handle dependencies between webppl headers should be order-independent, and would optimally be able to handle mutual dependencies as well, e.g. if `header1.wppl` also defined a variable `c` in terms of `b`

**Comment by stuhlmueller** [comment]
I think we'd be best off emulating node's `require` to make dependencies between `.wppl` (and also `.js`) files explicit. See my proposal [here](https://github.com/probmods/webppl/issues/211#issuecomment-142132326).

However, in the meantime, relying on the order in `package.json` doesn't seem terrible. Have you run into cases where this is a serious obstacle? (We should document it, though.)

**Comment by hawkrobe** [comment]
I agree that's the best solution (I feel uncomfortable calling functions that are defined in some .wppl header, since it's not obvious from reading the code which package they come from, or what header inside that package). 

My problem was just that I didn't realize the order mattered, so no I haven't run into any cases where it's a serious obstacle. It's just not obvious from the [package](http://docs.webppl.org/en/master/packages.html) documentation.
